src/pyodide.js

```diff
@@ -770,13 +772,42 @@ def temp(Module):
   // TODO: Should we have this?
   Module.globals = Module.wrapNamespace(Module.globals);
 
-  delete self.Module;
   let response = await fetch(`${baseURL}packages.json`);
   Module.packages = await response.json();
 
-  fixRecursionLimit(self.pyodide);
-  self.pyodide = makePublicAPI(self.pyodide, PUBLIC_API);
-  self.pyodide.registerJsModule("js", globalThis);
-  self.pyodide.registerJsModule("pyodide_js", self.pyodide);
-})();
-languagePluginLoader
+  fixRecursionLimit(Module);
+  let pyodide = makePublicAPI(Module, PUBLIC_API);
+  Module.registerJsModule("js", globalThis);
+  Module.registerJsModule("pyodide_js", pyodide);
+  globalThis.pyodide = pyodide;
+  return pyodide;
+};
+
+if (globalThis.languagePluginUrl) {
+  console.warn(
+      "languagePluginUrl is deprecated and will be removed in version 0.18.0," +
+      "instead use loadPyodide({ baseURL : <some_url>})");
```

Comment:
Currently I've set it up so that it *does* have an effect. These changes are backwards compatible except for the old case when people had left `languagePluginUrl` unset:
https://github.com/iodide-project/pyodide/blob/782a596d6f573f6a8d311b4afb2db5783e942701/src/pyodide.js#L812

Reply:
I meant that it doesn't when calling `loadPyodide` and setting languagePluginUrl. But since it's used for backward compatibility in languagePluginLoader you are right that we can't warn here.

Reply:
Okay I made `loadPyodide` throw an error if called twice and added a special warning if `languagePluginURL` is defined.